src/Microsoft.OData.Core/ODataMessageReader.cs

```diff
@@ -169,6 +169,17 @@ public ODataMessageReader(IODataResponseMessage responseMessage, ODataMessageRea
             this.model = model ?? GetModel(this.container);
             this.edmTypeResolver = new EdmTypeReaderResolver(this.model, this.settings.ClientCustomTypeResolver);
 
+            // Whether null values were omitted in the response.
+            this.settings.NullValuesOmitted = false;
+            string omitValuePreferenceApplied = responseMessage.PreferenceAppliedHeader().OmitValues;
+            if (omitValuePreferenceApplied != null)
+            {
+                // If the Preference-Applied header's omit-values parameter is present in the response, its value should
+                // be applied to the reader's setting.
+                this.settings.NullValuesOmitted =
+                    omitValuePreferenceApplied.Equals(ODataConstants.OmitValuesNulls, StringComparison.OrdinalIgnoreCase);
+            }

```

Comment:
The check for the omit-nulls preference in the ODataMessageWriter is more concise.  Any reason not to use the same code here?

Reply:
Updated to be the same as the writer; should be equivalent except that on reader side this flag should be false by default.

---
In reply to: [186526760](https://github.com/OData/odata.net/pull/1106#discussion_r186526760) [](ancestors = 186526760)